### PR TITLE
QManga

### LINKED ISSUE #14745: QManga
### Source name

qManga

### Source link

https://qmanga5.net/

### Language

Vietnamese

### Other details

NSFW? Yes

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by beerpiss**
Closes #14745

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/vi/qmanga/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="eu.kanade.tachiyomi.extension">
+	<application>
+		<activity
+			android:name="eu.kanade.tachiyomi.extension.vi.qmanga.QMangaUrlActivity"
```
Comment (AntsyLich) [inline src/vi/qmanga/AndroidManifest.xml:6]: Too much indents

File: src/vi/qmanga/src/eu/kanade/tachiyomi/extension/vi/qmanga/QManga.kt [hunk 2]
```diff
@@ -0,0 +1,1073 @@
...
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+class QManga : HttpSource() {
```
Comment (AntsyLich) [inline src/vi/qmanga/src/eu/kanade/tachiyomi/extension/vi/qmanga/QManga.kt:23]: Use `ParsedHttpSource`

**Review by AntsyLich — CHANGES_REQUESTED** [review]

File: src/vi/qmanga/src/eu/kanade/tachiyomi/extension/vi/qmanga/QManga.kt (hunk 2 above)
Comment (beerpiss) [reply src/vi/qmanga/src/eu/kanade/tachiyomi/extension/vi/qmanga/QManga.kt:23]: `popularMangaParse`, `latestUpdatesParse`, `searchMangaParse` all parse the same way. I intentionally used `HttpSource` to make it shorter and to not repeat a bunch of code.

File: src/vi/qmanga/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest
+	xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (alessandrojean) [inline src/vi/qmanga/AndroidManifest.xml:3]: As Lich said, fix the indentation and replace the tabs by spaces.

**Review by alessandrojean — CHANGES_REQUESTED** [review]

**Pull request merged**